sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpResponse.java

```diff
@@ -88,10 +72,30 @@ public final HttpRequest getRequest() {
         return request;
     }
 
+    /**
+     * Buffers the {@link HttpResponse}.
+     * <p>
+     * If {@link #isBuffered()} is true, this instance of {@link HttpResponse} is returned. Otherwise, a new instance
+     * of {@link HttpResponse} is returned where {@link #getBody()} is buffered into memory.
+     *
+     * @return The buffered {@link HttpResponse}.
+     */
+    public abstract HttpResponse buffer();
```

Comment:
do you think we can live without this method for now since now it's equivalent to getBody().toReplayableBinaryData() ?